### DISCUSSION TITLE
OData with minimal api in .NET 6

### DISCUSSION THREAD

**Discussion by danmendonca** [OData/AspNetCoreOData#801] [Q&A]
Hello,

I've been scratching my head over this, is it possible to have OData with an application using NET 6 minimal API? I have not seen a single working example with a minimal API, as far as I understood, the usage of controllers and the way attributes are set makes using OData with minimal APIs "impossible". Am I missing something? Is it possible?

**Comment by Schinwinkwinsky** [comment]
I'm not sure, but I believe OData has certain dependency to controllers.

**Comment by ElizabethOkerio** [comment]
@chrisspre

**Reply by xtofs** [reply]
the dependency on controllers is twofold as far as I understand it. 
a) the name of the controller has to match an entitySet/singleton
b)  it groups all operations for this  entitySet/singleton together so that they all can use the same DI services. 

b) can be dealt with in a straight forward way by providing a parameter for that service in each method and minimal API injects is that way. It is a repetition but not critical to "fix" from my point of view. 
a) is convenient and somewhat in the spirit of convention over configuration. But is for one something that is not in the spirit of minimal APIs and also somewhat error prone for beginners. If the Controller is not named properly, no error message at build or service start is shown and the request to that entitySet/singleton fails (5xx) without any diagnostics output.
It also has limitation that can be seen when contained navigation properties exist since there is "no good name" for the responsible controller. 

I could imaging that the mapping from entitySet/singleton to action methods can be done in a similar way that the ModelBuilder is configured: methods that "select" a model element (like and EntitySet/Singleton) and map it to action methods. 
Happy to work on a more detailed proposal @ElizabethOkerio  @mikepizzo

**Comment by habbes** [comment]
The library does not currently support minimal APIs, but this is an area we're exploring: cc @ElizabethOkerio
